### PR TITLE
make simple storage example app

### PR DESCRIPTION
**Pull request by martyall**
- [x] define message types in proto files
- [x] write application handlers
- [x] use logger(blocked by #19)
- ~~[ ] write tests using client lib (blocked by #31)~~

### CODE REVIEW

File: hs-abci-example/src/SimpleStorage/DB.hs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+              }
+
+-- | Get an item from the DB using its hash as a key.
+get
+  :: forall a name.
+     Binary a
+  => Connection name
+  -> ByteString
+  -> IO (Maybe a)
+get (Connection c) k = do
```
Comment (safareli) [inline hs-abci-example/src/SimpleStorage/DB.hs]: get c k = withHashTree c $ AT.lookup k

File: hs-abci-example/app/SimpleStorage/App.hs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+makeAndServeApplication = do
+  cfg <- makeAppConfig
+  let ioApp = transformApp (transformHandler cfg) $ app
+  serveApp =<< hookInMiddleware ioApp
+  where
+    mkMiddleware :: IO (Middleware IO)
+    mkMiddleware = do
+      logger <- mkLogStdoutDev
+      pure . appEndo . fold $
+        [ Endo logger
```
Comment (safareli) [inline hs-abci-example/app/SimpleStorage/App.hs:25]: What's the point of folding one element array?

**Review by charlescrain — APPROVED** [review]
Looks good

File: hs-abci-example/app/SimpleStorage/Handlers.hs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+checkTxH (Req.RequestCheckTx checkTx) = do
+  case decodeAppTxMessage $ checkTx ^. Req._checkTxTx . to convert of
+    Left e -> throwError $ DecodeTxError e
+    Right (ATMUpdateCount updateCountTx) -> do
+      AppConfig{countConnection} <- ask
+      eRes <- liftIO $ atomically $ do
+        let Connection c = countConnection
+        db <- readTVar c
+        pure $ stageTransaction db $
+          updateCount updateCountTx
```
Comment (safareli) [inline hs-abci-example/app/SimpleStorage/Handlers.hs:87]: AFAIK checktx shouldn't mutate state.

File: hs-abci-example/app/SimpleStorage/App.hs (hunk 2 above)
Comment (martyall) [reply hs-abci-example/app/SimpleStorage/App.hs:25]: there isn't one!

Comment (martyall) [reply hs-abci-example/app/SimpleStorage/App.hs:25]: But there will be, I'm going to be putting in some more middleware and I just wanted to leave it as a pattern for a demo with Kadena

**Pull request merged**